ui/src/dashboards/constants/gaugeColors.js

```diff
@@ -0,0 +1,103 @@
+export const MAX_THRESHOLDS = 5
+export const MIN_THRESHOLDS = 2
+
+export const COLOR_TYPE_MIN = 'min'
+export const DEFAULT_VALUE_MIN = '0'
+export const COLOR_TYPE_MAX = 'max'
+export const DEFAULT_VALUE_MAX = '100'
+export const COLOR_TYPE_THRESHOLD = 'threshold'
+
+export const GAUGE_COLORS = [
+  {
+    hex: '#BF3D5E',
+    name: 'ruby',
+  },
+  {
+    hex: '#DC4E58',
+    name: 'fire',
+  },
+  {
+    hex: '#F95F53',
+    name: 'curacao',
+  },
+  {
+    hex: '#F48D38',
+    name: 'tiger',
+  },
+  {
+    hex: '#FFB94A',
+    name: 'pineapple',
+  },
+  {
+    hex: '#FFD255',
+    name: 'thunder',
+  },
+  {
+    hex: '#7CE490',
+    name: 'honeydew',
+  },
+  {
+    hex: '#4ED8A0',
+    name: 'rainforest',
+  },
+  {
+    hex: '#32B08C',
+    name: 'viridian',
+  },
+  {
+    hex: '#4591ED',
+    name: 'ocean',
+  },
+  {
+    hex: '#22ADF6',
+    name: 'pool',
+  },
+  {
+    hex: '#00C9FF',
+    name: 'laser',
+  },
+  {
+    hex: '#513CC6',
+    name: 'planet',
+  },
+  {
+    hex: '#7A65F2',
+    name: 'star',
+  },
+  {
+    hex: '#9394FF',
+    name: 'comet',
+  },
+  {
+    hex: '#383846',
+    name: 'pepper',
+  },
+  {
+    hex: '#545667',
+    name: 'graphite',
+  },
+]
+
+export const DEFAULT_COLORS = [
+  {
+    type: COLOR_TYPE_MIN,
+    hex: GAUGE_COLORS[11].hex,
+    id: '0',
+    name: GAUGE_COLORS[11].name,
+    value: DEFAULT_VALUE_MIN,
+  },
+  {
+    type: COLOR_TYPE_MAX,
+    hex: GAUGE_COLORS[14].hex,
+    id: '1',
+    name: GAUGE_COLORS[14].name,
+    value: DEFAULT_VALUE_MAX,
+  },
+]
+
+export const validateColors = colors => {
+  const hasMin = colors.some(color => color.type === COLOR_TYPE_MIN)
+  const hasMax = colors.some(color => color.type === COLOR_TYPE_MAX)
+
+  return hasMin && hasMax
+}
```

Comment:
You need to check to see if colors is defined before you do anything else, otherwise return false. This prevents an uninitialized cell from being added. The error that is being created here will prevent me from reviewing the rest of the functionality. This is a PR blocker.